src/code/sys_rumble.c

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file sys_rumble.c
+ *
+ * This file implements a manager for storing and processing rumble pak requests made by the game state. Despite some
+ * parts of the system appearing to accommodate all four controller ports, only controller 1 will rumble according to
+ * the processed requests.
+ * This file is half of the system that runs on the padmgr thread alongside controller communications. The rest of the
+ * system that receives the requests from the game state runs on the graph thread and is implemented in `z_rumble.c`.
+ *
+ * @see RumbleMgr
+ * @see z_rumble.c
+ *
+ * @note Original filename is ikely sys_vibrate.c or similar as it is ordered after sys_ucode.c
+ */
+#include "global.h"
+
+/**
+ * Rumble manager update, runs on Vertical Retrace on the padmgr thread.
+ */
+void RumbleMgr_Update(RumbleMgr* rumbleMgr) {
+    static u8 wasEnabled = true;
+    s32 i;
+    s32 strength;
+    s32 index = -1;
+
+    // Clear enable status for all controllers
+    for (i = 0; i < MAXCONTROLLERS; i++) {
+        rumbleMgr->rumbleEnable[i] = false;
+    }
+
+    if (!rumbleMgr->updateEnabled) {
+        if (wasEnabled) {
+            // If it was previously enabled, reset pak type
+            for (i = 0; i < MAXCONTROLLERS; i++) {
+                gPadMgr.pakType[i] = CONT_PAK_NONE;
+            }
+        }
+        wasEnabled = rumbleMgr->updateEnabled;
+        return;
+    }
+
+    wasEnabled = rumbleMgr->updateEnabled;
```

Comment:
I'm puzzled by what this achieves

The only thing that sets `updateEnabled=false` afaict is `Rumble_SetUpdateEnabled(false);` in `Play_Update`, which should be reached every frame while in pause menu/game-over

I think `pakType[i] = CONT_PAK_NONE` has the purpose of entering the `if (!triedRumbleComm) {` block at the end of `PadMgr_UpdateRumble` ?

But that `if (!triedRumbleComm) {` block has
```c
osMotorStart(...);
osMotorStop(...);
```

Does that make the controller rumble for a fraction of a second when pausing?

And what is otherwise the idea with this? Re-attempt rumble comm on pausing or dying?

Reply:
I don't think the intention is to re-init rumble on pausing and dying, I think the intention ends at stopping rumble from updating. The rest is just a side-effect due to padmgr doing its own thing. I don't think it's intentional but I also don't know if you can tell this is happening.

Reply:
The early return here is what stops rumble from updating, right?

Specifically `pakType[i] = CONT_PAK_NONE;`, what would that be useful for?